### PR TITLE
Allow folder-structure to be none

### PR DESCRIPTION
**Pull request by sinaxin**
When set to none, allows to download all photos to base directory.

### CODE REVIEW

**Comment by AndreyNikiforov** [comment]
@sinaxin is it possible to add unit test for your new feature?

File: tests/test_folder_structure.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                    "-d",
+                    "tests/fixtures/Photos",
+                ],
+            )
+            print_result_exception(result)
+            filenames = result.output.splitlines()
+
+            self.assertEqual(len(filenames), 8)
+            self.assertEqual(
+                "tests/fixtures/Photos/2018/07/31/IMG_7409.JPG", filenames[0]
```
Comment (AndreyNikiforov) [inline tests/test_folder_structure.py:45]: Please use normpath to allow tests to be executed on other platforms, e.g. Windows. See other tests as an example.

**Review by AndreyNikiforov — CHANGES_REQUESTED** [review]
Please add a note about your new feature to CHANGELOG.md file -- that file will be used for release notes.

**Comment by menkej** [comment]
Wow! Whatever happened now! I was just wanting to rebase this branch to include the Changelog and stuff. Now it includes a hell of a lot of other changes as well...

This was basically my first rebase (not the git expert yet). Is this just "first time hurts" but fine anyways? Or did I mess things up now?

**Comment by AndreyNikiforov** [comment]
> Wow! Whatever happened now! I was just wanting to rebase this branch to include the Changelog and stuff. Now it includes a hell of a lot of other changes as well...
> 
> This was basically my first rebase (not the git expert yet). Is this just "first time hurts" but fine anyways? Or did I mess things up now?

@menkej why are you pushing to @sinaxin branch? 

I usually merge master into branch, not rebase, as I prefer my commits to be immutable. I do not have much experience with rebases in GitHub. If the final state of the branch matches what you expected to see after rebase, then you are good and don't worry about many commits. ;-)

**Comment by AndreyNikiforov** [comment]
@menkej alternative is to start over: new branch from master, cherry pick changes for the PR, then reset `patch-4` branch to the top of new branch, and force push `patch-4`.

**Comment by menkej** [comment]
> I usually merge master into branch

This was basically my intention - the CHANGELOG was missing on that branch. I wanted to use the initial pull-request branch to honour @sinaxin 's effort. But as there was no further activity I decided to just add the missing parts myself. I know that most of the time the test cases are quite a hurdle and I had an idea to test it.

I think the safest now is to reject this pull request, ditch patch-4 and create a new branch/pull request (just copy my changed files). Before I start cherrypicking and stuff. I'm still quite a newbie to git/github...

Or do you have objections @sinaxin ?

**Review by AndreyNikiforov — APPROVED** [review]
@menkej there are so many changes and it is hard to distinguish between merging master and changes for the 'none folder structure'. May be this is because of the merging master instead of rebasing (I assume you did merge). We can proceed now, but will have to solve that challenge with humongous diffs in other PR soon, otherwise reviewing will be hard/error prone.

**Comment by menkej** [comment]
> @menkej there are so many changes and it is hard to distinguish between merging master and changes for the 'none folder structure'. 

This is why I created #195 as an replacement for this! I'd ditch #169 and use #195. Could you have a look at that PR, please?

**Comment by menkej** [comment]
Closing in favour of #195

**Pull request closed without merging**